docs/userguide/design.md

```diff
@@ -840,6 +843,22 @@ libraries that resulted from the `–-disable-renaming` ICU build
 beginning with ICU version 3.0. Older versions of ICU (2.8 and earlier) do not
 provide for binary compatibility between versions.
 
+**Provide both “common” and “i18n” libraries, or build a combined library.**
+It is rare but possible that services/APIs move from one library to another.
+For example, many years ago we moved the BreakIterator APIs from i18n to common,
+so that word titlecasing functions no longer needed separate code to find
+titlecasing or word break opportunities.
+
+More recently, the ListFormatter moved from the common library to i18n
+when its features grew beyond primitive patterns to also support
+FieldPosition and FormattedValue features.
+
+There is also a third, “io” library.
+It is possible that some of its functionality may be moved to the i18n or common
+libraries.
+(A likely candidate might be `operator<<(std::ostream& stream, const UnicodeString& s)`
+although there are no actual plans to do so at the time of this writing.)
+
```

Comment:
Thanks Markus.

I wonder if it would be possible to add the following (or something like it) in order to add some emphasis on why this can be rather problematic on some platforms:

> One can build a combined library which provides the exports from both the “common” and “i18n” libraries, in order to provide a single library for linking against.
>
> This may be needed for some platforms where there is a strong relationship between an API and the library that implements it. For example, on Windows platforms, attempting to find an API that has been moved with a `LoadLibrary`/`GetProcAddress` approach will fail, unless you are using a combined library.

Reply:
sgtm, amended

Reply:
thx